eventservice/api/utils/group_auth.py

```diff
@@ -61,6 +61,17 @@ def get_group_memberships():
 
 # This util method checks if a user has admin permissions in a group event
 def check_group_event_admin_access(event, group_memberships):
+    if event and event.get('groupIds'):
+        if len(event.get('groupIds')) != len(group_memberships):
+            return False
+        for groupId in event.get('groupIds'):
+            found = False
+            for group_member in group_memberships:
+                if groupId == group_member.get('id') and group_member.get('role') == 'admin':
+                    found = True
+                    break
+        if not found:
+            return False
     if event and event.get('createdByGroupId'):
```

Comment:
The logic here is : 
If (the event belongs to groupA, and the user has admin access to groupA) and (event is created by groupA, and user is admin of groupA) -> return True.

Can the below situation arise in group events ? : 
Event1 is created by groupA. Event1 is also part of groupB. User1 is admin of groupB, but only a member of groupA.
event1.get(groupIds) will return [groupA, groupB]. Event1.get(createdByGroupId) returns ['groupA']. 
In this situation, the code below will return False.
```
        for groupId in event.get('groupIds'):
            found = False
            for group_member in group_memberships:
                if groupId == group_member.get('id') and group_member.get('role') == 'admin':
                    found = True    # we have found=True here and we break.
                    break
    if event and event.get('createdByGroupId'):
        found = False
        for group_member in group_memberships:
            if event.get('createdByGroupId') == group_member.get('id') and group_member.get('role') == 'admin':
                found = True
                break
        if not found:
            return False  # we return False here.
```

Just wanted to clarify if this is the logic wanted.

Reply:
it requires the user has the all groups admin.